app/models/candidate.rb

```diff
@@ -28,7 +28,8 @@ def self.for_email(email)
 
   def current_application
     application_form = application_forms.order(:created_at).last
-    application_form ||= application_forms.create!
+    application_form ||= application_forms.create! if Time.zone.now < CycleTimetable.apply_1_deadline
+    application_form ||= application_forms.create!(recruitment_cycle_year: CycleTimetable.next_year) if Time.zone.now > CycleTimetable.apply_1_deadline
```

Comment:
Would it be clearer if the above two lines were refactored into an `if ... else` or `? :`? There is also the (razor thin) edge case where `Time.zone.now == CycleTimetable.apply_1_deadline` that this doesn't cover.

Reply:
Yea an if/else 100% makes sense here. Not really sure why i went with this 🤔